packages/kit/src/context.ts

```diff
@@ -1,9 +1,22 @@
-import { getContext } from 'unctx'
+import { AsyncLocalStorage } from 'node:async_hooks'
+import { createContext, getContext } from 'unctx'
 import type { Nuxt } from '@nuxt/schema'
 
-/** Direct access to the Nuxt context - see https://github.com/unjs/unctx. */
+/**
+ * Direct access to the Nuxt global context - see https://github.com/unjs/unctx.
+ * @deprecated Use `getNuxtCtx` instead
+ */
 export const nuxtCtx = getContext<Nuxt>('nuxt')
 
+/** async local storage for the name of the current nuxt instance */
+const asyncNameStorage = createContext({
+  asyncContext: true,
+  AsyncLocalStorage,
+})
+
+/** Direct access to the Nuxt context with asyncLocalStorage - see https://github.com/unjs/unctx. */
+export const getNuxtCtx = () => getContext<Nuxt>(asyncNameStorage.tryUse())
```

Comment:
@pi0 is it okay to use `getContext` with an undefined value with `tryUse()`?

Reply:
It seems you are accessing by name (why?)

nuxt context instances are themselves unique. I should be something like this in impl:

- `const nuxtCtx = useContext('nuxt', { .. })`
- `getNuxtCtx = () => nuxtCtx.use()
- `runWithNuxtContext = (nuxt, fn) => { nuxtCtx.call(nuxt, fn) }`

Reply:
I think i did taht to allow users to use `getContext('nuxt_name')` outside of any instance, just in case (maybe the storybook module will need this).

Do you prefer to have the instance of nuxt being stored ?

Reply:
right :+1: , let's go for the nuxt instance being stored. We can open another PR to add this feature